sor/src/test/java/com/bazaarvoice/emodb/sor/core/DataStoreTest.java

```diff
@@ -189,6 +176,190 @@ private void doTest(DataStore store) throws Exception {
+        store.update(TABLE, KEY1, changeId, Deltas.mapBuilder().put("key", "value1").build(), audit, WriteConsistency.STRONG);
+        record = store.get(TABLE, KEY1);
+        // Only the last update date should have changed
+        lastUpdateDate = now;
+        verifyContentAndTimestamps(record, "value1", firstUpdateDate, lastMutateDate, lastUpdateDate);
+
+        // Make a mutative change
+        now = new Date(now.getTime() + 1000);
+        changeId = com.bazaarvoice.emodb.common.uuid.TimeUUIDs.uuidForTimestamp(now);
+        store.update(TABLE, KEY1, changeId, Deltas.mapBuilder().put("key", "value2").build(), audit, WriteConsistency.STRONG);
+        record = store.get(TABLE, KEY1);
+        // Both last update and mutate dates should have changed
+        lastMutateDate = lastUpdateDate = now;
+        verifyContentAndTimestamps(record, "value2", firstUpdateDate, lastMutateDate, lastUpdateDate);
+    }
+
+    @Test
+    public void testRecordTimestampsWithEventTags() throws Exception {
+        DataStore store = new InMemoryDataStore(new MetricRegistry());
+
+        TableOptions options = new TableOptionsBuilder().setPlacement("default").build();
+        assertFalse(store.getTableExists(TABLE));
+        store.createTable(TABLE, options, Collections.<String,Object >emptyMap(), newAudit("create table"));
+        assertTrue(store.getTableExists(TABLE));
+
+        // Record does not exist initially, so all dates should not be present
+        Map<String, Object> record = store.get(TABLE, KEY1);
+        assertTrue(Intrinsic.isDeleted(record));
+        assertNull(Intrinsic.getFirstUpdateAt(record));
+        assertNull(Intrinsic.getLastUpdateAt(record));
+        assertNull(Intrinsic.getLastMutateAt(record));
+
+        Audit audit = newAudit("test");
+
+        Date now = new Date(System.currentTimeMillis() - 5000);
+        // Normally using uuidForTimestamp is a bad idea, but for this test we need precise control over the UUID timestamps.
+        // So long as no two are created with the same time we'll be ok.
+        UUID changeId = com.bazaarvoice.emodb.common.uuid.TimeUUIDs.uuidForTimestamp(now);
+        store.update(TABLE, KEY1, changeId, Deltas.mapBuilder().put("key", "value0").build(), audit, WriteConsistency.STRONG);
+        Date firstUpdateDate = now;
+        // Perform a mutative update
+        now = new Date(now.getTime() + 1000);
+        changeId = com.bazaarvoice.emodb.common.uuid.TimeUUIDs.uuidForTimestamp(now);
+        store.update(TABLE, KEY1, changeId, Deltas.mapBuilder().put("key", "value1").build(), audit, WriteConsistency.STRONG);
+        Date lastContentMutateDate = now;
+        // Perform a non-mutative update
+        now = new Date(now.getTime() + 1000);
+        changeId = com.bazaarvoice.emodb.common.uuid.TimeUUIDs.uuidForTimestamp(now);
+        store.update(TABLE, KEY1, changeId, Deltas.mapBuilder().put("key", "value1").build(), audit, WriteConsistency.STRONG);
```

Comment:
May want to use `no-op` delta for non-mutative changes to make it perfectly clear that these changes are not going to change the content in any way.